src/main/java/com/google/cloud/tools/app/deploy/gcloud/GcloudAppDeploy.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.app.deploy.gcloud;
+
+import com.google.cloud.tools.app.api.AppEngineException;
+import com.google.cloud.tools.app.api.deploy.DeployConfiguration;
+import com.google.cloud.tools.app.deploy.Deploy;
+import com.google.cloud.tools.app.deploy.process.CliProcessStarter;
+import com.google.cloud.tools.app.deploy.process.LoggingOutputHandler;
+import com.google.cloud.tools.app.deploy.process.OutputHandler;
+import com.google.cloud.tools.app.deploy.process.ProcessStarter;
+import com.google.cloud.tools.app.deploy.process.StreamConsumer;
+import com.google.cloud.tools.app.impl.cloudsdk.internal.sdk.CloudSdk;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.concurrent.Future;
+import java.util.logging.Level;
+
+public class GcloudAppDeploy implements Deploy {
+  private final CloudSdk sdk;
+  private final ConfigurationTranslator<DeployConfiguration> configurationTranslator;
+  private OutputHandler outputHandler;
+  private ProcessStarter processStarter;
+
+  private GcloudAppDeploy(CloudSdk sdk,
+      ConfigurationTranslator<DeployConfiguration> configurationTranslator,
+      ProcessStarter processStarter) {
+    this.sdk = sdk;
+    this.processStarter = processStarter;
+    this.configurationTranslator = configurationTranslator;
+  }
+
+  @Override
+  public Future<String> deploy(DeployConfiguration config) {
+    try {
+      String[] command = sdk.createAppCommand("deploy", configurationTranslator.translate(config));
+      Process process = processStarter.startProcess(command);
+
+      if (outputHandler == null) {
+        outputHandler = new LoggingOutputHandler(GcloudAppDeploy.class.getName(), Level.INFO);
+      }
+      StreamConsumer.startNewConsumer(process.getErrorStream(), outputHandler);
+
+      return new GcloudFuture(process);
+
+    } catch(IOException e) {
```

Comment:
Checked exceptions should not be converted to runtime exceptions. This operation cries out for a checked exception..